bin/get_metadata.py

```diff
@@ -0,0 +1,113 @@
+#!/usr/bin/env python
+from __future__ import print_function
+import xmltodict
+from collections import OrderedDict
+import re
+import argparse
+import os
+import json
+
+
+class RunfolderInfo():
+
+    def __init__(self, runfolder):
+        self.runfolder = runfolder
+        self.run_parameters = self.read_run_parameters()
+        self.stats_json = self.read_stats_json()
+        self.description_and_identifier = OrderedDict()
+        self.run_parameters_tags = \
+            {'RunId': 'Run ID', 'RunID': 'Run ID',
+             'ApplicationName': 'Control software', 'Application': 'Control software',
+             'ApplicationVersion': 'Control software version',
+             'Flowcell': 'Flowcell type', 'FlowCellMode': 'Flowcell type',
+             'ReagentKitVersion': 'Reagent kit version',
+             'RTAVersion': 'RTA Version', 'RtaVersion': 'RTA Version',
+             }
+
+    def find(self, d, tag):
+        if tag in d:
+            yield d[tag]
+        for k, v in d.items():
+            if isinstance(v, dict):
+                for i in self.find(v, tag):
+                    yield i
+
+    def read_run_parameters(self):
+        alt_1 = os.path.join(self.runfolder, "runParameters.xml")
+        alt_2 = os.path.join(self.runfolder, "RunParameters.xml")
+        if os.path.exists(alt_1):
+            with open(alt_1) as f:
+                return xmltodict.parse(f.read())
+        elif os.path.exists(alt_2):
+            with open(alt_2) as f:
+                return xmltodict.parse(f.read())
+        else:
+            return None
+
+    def read_stats_json(self):
+        stats_json_path = os.path.join(
+            self.runfolder, "Unaligned/Stats/Stats.json")
```

Comment:
I think it would make sense to not hard code this path, since `Unaligned` is not the `bcl2fastq` default path. Perhaps it could be passed as an argument on the command line?

Reply:
Very true, adding that.